tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/LambdaHandler.cs

```diff
@@ -0,0 +1,57 @@
+// <copyright file="LambdaHandler.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Reflection;
+
+namespace Datadog.Trace.ClrProfiler.ServerlessInstrumentation
+{
+    internal class LambdaHandler
+    {
+        private string[] handlerTokens;
+
+        internal LambdaHandler(string handlerName)
+        {
+            string[] stringSeparators = new string[] { "::" };
+            handlerTokens = handlerName.Split(stringSeparators, StringSplitOptions.None);
+        }
+
+        internal string GetAssembly()
+        {
+            return handlerTokens[0];
+        }
+
+        internal string GetFullType()
+        {
+            return handlerTokens[1];
+        }
+
+        internal string GetMethodName()
+        {
+            return handlerTokens[2];
+        }
+
+        internal string[] BuidParamTypeArray()
+        {
+            Type myClassType = Type.GetType(GetTypeName());
+            MethodInfo customerMethodInfo = myClassType.GetMethod(GetMethodName());
```

Comment:
small nit: `Type.GetType` and `myClassType.GetMethod` can return null, and that exception is not being handled when calling `BuildLambdaHandlerDefinition`.

Reply:
thanks! the error is now caught in `GetServerlessDefinitions`

Reply:
If `Type.GetType()` returns null here, for example, the log will only have the following message, which won't be very helpful:
```
Impossible to get Serverless Definitions : Object reference not set to an instance of an object.
```

Ideally, we should check for null here, log a better message, and avoid the exception entirely. Maybe something like this (note the use of the null-coalescing operator, `?.`, in the 2nd and 3rd lines):
```csharp
Type myClassType = Type.GetType(GetTypeName());
MethodInfo customerMethodInfo = myClassType?.GetMethod(GetMethodName());
ParameterInfo[] methodParameters = customerMethodInfo?.GetParameters();

if(methodParameters == null)
{
    Serverless.Error($"Could not get method parameters for {GetTypeName()}.{GetMethodName()} in {GetAssembly()}.";
    return Array.Empty<string>();
}

string[] paramType = new string[methodParameters.Length + 1];
paramType[0] = customerMethodInfo.ReturnType.Name;
for (var i = 0; i < methodParameters.Length; i++)
{
	paramType[i + 1] = methodParameters[i].ParameterType.ToString();
}

return paramType;
```

Reply:
This doesn't need to block this PR, but maybe something to clean up later before GA?